frontend/catalyst/compiler.py

```diff
@@ -264,6 +264,11 @@ def get_default_flags(options):
         mlir_lib_path = get_lib_path("llvm", "MLIR_LIB_DIR")
         rt_lib_path = get_lib_path("runtime", "RUNTIME_LIB_DIR")
 
+        sys.path.append(get_lib_path("runtime", "RUNTIME_LIB_DIR"))
+        import catalyst_callback_registry as registry  # pylint: disable=import-outside-toplevel
+
+        registry.set_mlir_lib_path(mlir_lib_path)
```

Comment:
Not sure to follow why it is necessary now? Particularly the first line, could explain the reasoning here?

Reply:
I'll add the following comments (keeping this open until I commit them)
```python

        # Adds RUNTIME_LIB_DIR to the Python system path to allow the catalyst_callback_registry
        # to be importable.
        sys.path.append(get_lib_path("runtime", "RUNTIME_LIB_DIR"))

        import catalyst_callback_registry as registry  # pylint: disable=import-outside-toplevel

        # We use MLIR's C runner utils library in the registry.
        # In order to be able to dlopen that library we need to know the path
        # So we set the path here.
        registry.set_mlir_lib_path(mlir_lib_path)
```

Reply:
All good thanks :+1:

Reply:
https://github.com/PennyLaneAI/catalyst/pull/610/commits/df8779ec305ca7a86269f8b27c10c444ae735106